File: Daily Test/Python/Submatrix Sum 2.3.2026.py
s, r, c = map(int, input().split())

m = [list(map(int, input().split())) for _ in range(r)]
res = []

for top in range(r):
    for left in range(c):
        for bottom in range(top, r):
            for right in range(left, c):
                t = 0
                d = []
                for i in range(top, bottom + 1):
                    w = []
                    for j in range(left, right + 1):
                        t += m[i][j]
                        w.append(m[i][j])
                    d.append(w)
                if t == s:
                    res.append(d)

for k in range(len(res)):
    for row in res[k]:
        print(*row)
    if k != len(res) - 1:
        print("END")

File: Daily Test/C/Submatrix Sum 2.3.2026.c
#include <stdio.h>

int main() {
    int S, r, c;
    scanf("%d %d %d", &S, &r, &c);

    int m[r][c];

    for(int i = 0; i < r; i++)
        for(int j = 0; j < c; j++)
            scanf("%d", &m[i][j]);

    for(int top = 0; top < r; top++) {
        for(int left = 0; left < c; left++) {
            for(int bottom = top; bottom < r; bottom++) {
                for(int right = left; right < c; right++) {

                    int sum = 0;

                    for(int i = top; i <= bottom; i++)
                        for(int j = left; j <= right; j++)
                            sum += m[i][j];

                    if(sum == S) {
                        for(int i = top; i <= bottom; i++) {
                            for(int j = left; j <= right; j++)
                                printf("%d ", m[i][j]);
                            printf("\n");
                        }
                        printf("END\n");
                    }
                }
            }
        }
    }

    return 0;
}